snapcraft/pluginhandler.py

```diff
@@ -599,18 +612,29 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
         common.link_or_copy(src, dst, follow_symlinks=follow_symlinks)
 
 
+def _remove_empty_tree(directory, barrier):
+    """Remove an empty directory tree up to (not including) a barrier path."""
+
+    while (directory != barrier and os.path.isdir(directory) and
```

Comment:
There's of course the `listdir()` and the `rmdir` versus `rmtree`, but yeah it makes you blink. I'm not sure there's a better way to handle this though-- I found the problem with [this test case](https://github.com/ubuntu-core/snapcraft/pull/442/files/23c36316a7762b89e1809580c0bab05efbd154bc#diff-b308bc3f4f6ea8bc23d67863c6b31a33R803).